src/components/OSCALMetadata.js

```diff
@@ -325,12 +333,55 @@ export function OSCALMetadataParty(props) {
   );
 }
 
+export function OSCALMetadataRole(props) {
+  const [open, setOpen] = React.useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <Card>
+      <CardHeader subheader={props.role.title} />
+      <CardActions>
+        <Button
+          size="small"
+          variant="outlined"
+          onClick={handleOpen}
+          aria-label={`${props.role.title} details button`}
+        >
```

Comment:
```suggestion
        <Button
          size="small"
          variant="outlined"
          onClick={handleOpen}
          aria-label={`${props.role.title} details button`}
          disabled={!props.role?.description}
        >
```

Maybe we can just not allow opening the dialog if there isn't a description? Seems silly to get a confusing modal without text